src/shared/hooks/useDownload.js

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import fileSaver from 'file-saver';
+
+function useDownload() {
+  const [loading, setLoading] = React.useState(false);
+
+  async function downloadFile(url, name) {
+    setLoading(true);
+
+    const fileName = name || url.split('/').pop();
+
+    fetch(url)
+      .then(response => response.blob())
+      .then(blob => {
```

Comment:
You should also catch any exceptions that occur in the `fetch`

Reply:
done